Client/game_sa/CPlaceableSA.h

```diff
@@ -0,0 +1,34 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *
+ *  Multi Theft Auto is available from http://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+#include <CMatrix_Pad.h>
+#include <CVector.h>
+
+class CSimpleTransformSAInterface            // 16 bytes
+{
+public:
+    CVector m_translate;
+    float   m_heading;
+};
+
+class CPlaceableSAInterface
+{
+public:
+    virtual void* Destructor(bool free) = 0;
+
+    bool IsPlaceableVTBL() const { return *reinterpret_cast<const void* const*>(this) == (void*)0x863C40; }
+    bool HasVTBL() const { return *reinterpret_cast<const void* const*>(this) != nullptr; }
```

Comment:
```cpp
bool HasVTBL() const noexcept { return this != nullptr; }
```
?

Reply:
No, you check is different

Reply:
How is it different? You cast this pointer to the const void pointer essentially removing the type.

Reply:
This code checks if first object member (VTBL in virtual classes) is null. I think it means that an object wasn't initialized.
Your check has no sense